pesacheck_meedan_bridge/py/check_api.py

```diff
@@ -69,9 +70,11 @@ def post_to_check(data):
     body = dict(query=query)
     url = settings.PESACHECK_CHECK_URL
     response = requests.post(url, headers=headers, json=body, timeout=60)
+    res = response.json()
+    if res.get("errors"):
+        sentry_sdk.capture_exception(Exception(str(res.get("errors"))))
+        return None
     if response.status_code == 200:
```

Comment:
Not sure I fully follow this sequence here: Can res have errors while status_code is 200? If not, shouldn't that be the first thing we check for?

Reply:
@kilemensi  res can be status code 200 but with errors that's why I am handling any errors before the status code
![image](https://github.com/CodeForAfrica/api/assets/39160236/1d0b4e90-25f7-418d-8664-f88a133bd07b)

Reply:
Put another way, can a _correct_ query return status_code 200 but with errors?

Reply:
A correct query cannot return errors

Reply:
A correct query with errors will return a 4xx status code . @kilemensi

Reply:
Cool @koechkevin. Is it possible for us to send _incorrect_ query? If yes, under what conditions or why are we not handling those conditions?

Reply:
@kilemensi we have handled them however an edge case we haven't handled might occur(like the the bad unicode character issue we had with URLs).So I think it is safer to just handle the errors before checking the status code.

Reply:
Could be @koechkevin but I'm still no 100% sure with this approach here.

1. make a `request.post`: Can this throw exception i.e. does it need `try`/`except`?
2. the request.post returns:

You seem to be checking for errors here but shouldn't we check if `status_code` is `200` first _and then_ if there are `errors`? I mean if `status_code != 200`, isn't the error message in `response.text`?

But having said all of this, doesn't the  `try`/`except` in the `main.py` supposed to capture any edge cases we haven't explicitly handled? Isn't 👇🏽 be enough?

```python
    res = response.json()
    if response.status_code == 200 and res.get("data"):
        return res
    
    message = res.get("errors") or response.text
    raise Exception(message) # main.py will capture and send to sentry
```

Reply:
> # main.py will capture and send to sentry
I prefer sending to sentry at this point because this  function is called inside a for loop so it would exit a for loop if it returns an error.

Reply:
Cool @koechkevin but you do know that you can have a try/except in the loop, right? No need to "dirty" the code with sentry capture everywhere.